File: leetcode/src/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        result = []

        intervals.sort(key=lambda x:x[0])

        current_interval = intervals[0]

        for interval in intervals:
            if interval[0] <= current_interval[1]:
                current_interval[1] = max(current_interval[1], interval[1])
                continue
            result.append(current_interval)
            current_interval = interval
        result.append(current_interval)

        return result
        

File: leetcode/src/merge-intervals.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), [](vector<int>& a, vector<int>&b){
            return a[0] < b[0];
        });

        vector<vector<int>> result;
        vector<int> current_interval = intervals[0];

        for (const vector<int>& interval : intervals) {
            if (interval[0] <= current_interval[1]) {
                current_interval[1] = max(current_interval[1], interval[1]);
                continue;
            }
            result.push_back(current_interval);
            current_interval = interval;
        }
        result.push_back(current_interval);
        
        return result;
    }
};